The job **`mingw-check`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/9949948769/job/27487027267) [(plain)](https://github.com/rust-lang/rust/commit/21d7eba0842f6d3bf27077c58a3289dd8a81e63d/checks/27487027267/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
#12 [ 5/11] RUN npm install es-check@6.1.1 eslint@8.6.0 -g
#12 5.326 npm WARN deprecated inflight@1.0.6: This module is not supported, and leaks memory. Do not use it. Check out lru-cache if you want a good and tested way to coalesce async requests by a key value, which is much more comprehensive and powerful.
#12 5.388 npm WARN deprecated inflight@1.0.6: This module is not supported, and leaks memory. Do not use it. Check out lru-cache if you want a good and tested way to coalesce async requests by a key value, which is much more comprehensive and powerful.
#12 5.389 npm WARN deprecated rimraf@3.0.2: Rimraf versions prior to v4 are no longer supported
#12 5.443 npm WARN deprecated @humanwhocodes/config-array@0.9.5: Use @eslint/config-array instead
#12 5.468 npm WARN deprecated glob@7.2.3: Glob versions prior to v9 are no longer supported
#12 5.476 npm WARN deprecated @humanwhocodes/object-schema@1.2.1: Use @eslint/object-schema instead
#12 6.522 
#12 6.522 added 205 packages in 6s
#12 6.522 
#12 6.523 20 packages are looking for funding
---
#16 3.782 Building wheels for collected packages: reuse
#16 3.783   Building wheel for reuse (pyproject.toml): started
#16 4.140   Building wheel for reuse (pyproject.toml): finished with status 'done'
#16 4.141   Created wheel for reuse: filename=reuse-1.1.0-cp310-cp310-manylinux_2_35_x86_64.whl size=181117 sha256=f5f58750481f69515c2c0d1d503daf565e2565c370d07fc6aeb95fe3498b4269
#16 4.141   Stored in directory: /tmp/pip-ephem-wheel-cache-yjtjoq_l/wheels/c2/3c/b9/1120c2ab4bd82694f7e6f0537dc5b9a085c13e2c69a8d0c76d
#16 4.144 Installing collected packages: boolean-py, binaryornot, setuptools, reuse, python-debian, markupsafe, license-expression, jinja2, chardet
#16 4.169   Attempting uninstall: setuptools
#16 4.169     Found existing installation: setuptools 59.6.0
#16 4.171     Not uninstalling setuptools at /usr/lib/python3/dist-packages, outside environment /usr
---
   Compiling std v0.0.0 (/checkout/library/std)
error[E0428]: the name `strlen` is defined multiple times
   --> library/core/src/ffi/c_str.rs:751:1
    |
738 |     fn strlen(s: *const c_char) -> usize;
    |     ------------------------------------- previous definition of the value `strlen` here
...
751 | const unsafe fn strlen(ptr: *const c_char) -> usize {
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `strlen` redefined here
    |
    = note: `strlen` must be defined only once in the value namespace of this module

error[E0015]: cannot call non-const fn `c_str::strlen` in constant functions
    |
    |
285 |         let len = unsafe { strlen(ptr) };
    |
    = note: calls in constant functions are limited to constant functions, tuple structs and tuple variants

Some errors have detailed explanations: E0015, E0428.
```

</details>

--- a/library/core/src/ffi/c_str.rs
+++ b/library/core/src/ffi/c_str.rs
@@ -780,8 +780,15 @@ const unsafe fn const_strlen(ptr: *const c_char) -> usize {
 pub struct Bytes<'a> {
     // since we know the string is nul-terminated, we only need one pointer
     ptr: NonNull<u8>,
-    phantom: PhantomData<&'a u8>,
+    phantom: PhantomData<&'a [c_char]>,
 }
+
+#[unstable(feature = "cstr_bytes", issue = "112115")]
+unsafe impl Send for Bytes<'_> {}
+
+#[unstable(feature = "cstr_bytes", issue = "112115")]
+unsafe impl Sync for Bytes<'_> {}
+
 impl<'a> Bytes<'a> {
     #[inline]
     fn new(s: &'a CStr) -> Self {
@@ -814,7 +821,7 @@ impl Iterator for Bytes<'_> {
             if ret == 0 {
                 None
             } else {
-                self.ptr = self.ptr.offset(1);
+                self.ptr = self.ptr.add(1);
                 Some(ret)
             }
         }
@@ -823,6 +830,12 @@ impl Iterator for Bytes<'_> {
     #[inline]
     fn size_hint(&self) -> (usize, Option<usize>) {
         if self.is_empty() { (0, Some(0)) } else { (1, None) }
+    }
+
+    #[inline]
+    fn count(self) -> usize {
+        // SAFETY: We always hold a valid pointer to a C string
+        unsafe { strlen(self.ptr.as_ptr().cast()) }
     }
 }
 

